ingest/src/main/java/com/microsoft/azure/kusto/ingest/IngestClientImpl.java

```diff
@@ -29,16 +34,23 @@
 class IngestClientImpl implements IngestClient {
 
     private final static Logger log = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
-
     private static final int COMPRESSED_FILE_MULTIPLIER = 11;
     private final ResourceManager resourceManager;
     private AzureStorageClient azureStorageClient;
+    private String connectionDataSource;
+    private String endpointServiceType;
+    private String suggestedEndpointUri;
+    private static final String INGEST_PREFIX = "ingest-";
+    protected static final String WRONG_ENDPOINT_MESSAGE =
+            "You are using '%s' client type, but the provided endpoint is of ServiceType '%s'. Initialize the client with the appropriate endpoint URI";
+    public static final String DATA_MANAGEMENT_SERVICE_TYPE = "DataManagement";
```

Comment:
Because we only need 1 value.

Reply:
What about a case where a customer is trying to use streaming ingest with DM endpoint? we'll need to add "Engine" as well.

Reply:
Apparently that's not something that happens often. People by default use their Engine endpoint. The confusing part is realizing that the "ingest-" prefix needs to be added to the DM endpoint.

We added it to both sides in the .NET SDK because it offers ingestion clients for both DM (Queued) and the Engine (Direct). The Java SDK only offers DM ingestion, so there's little risk of confusion, and apparently doesn't happen enough in practice to be worth adding.

Reply:
I'll let @vladikbr explain the rationale further to what I wrote above if he'd like.

Reply:
We do have the streaming ingest client here, don't we?
So it does make sense to add the check to the streaming client as well (for the other way around).
BTW, this class needs to be renamed to QueuedIngestClient and made public, similar to StreamingIngestClient

Reply:
As discussed, adding wrong endpoint detection to both a Queued and Streaming ingest client involved either duplicated code, or restructuring more than seems appropriate for such a feature.
This is because there’s no common Impl parent shared by QueuedIngestClient and StreamingIngestClient. Likewise, the ClientImpl implements both the Client and StreamingClient Interfaces, but there’s not shared parent Impl. Also, the QueuedIngestClient executes commands using the ResourceManager, and the StreamingIngestClient executes commands directly using the client.

In determining whether such duplication/restructuring would be worth it, we considered that we didn’t add wrong endpoint detection to the StreamingIngestClient in .NET (just Queued and Direct), and it's very rare to encounter a wrong endpoint for a StreamingIngest user. People who use StreamingIngestion have several things to first set up.

Therefore, we are adding this wrong endpoint detection for the Streaming client feature in a separate PR, which we'll evaluate separately whether it's worth adding: https://github.com/Azure/azure-kusto-java/pull/149